### PR TITLE
feat: enable CODEOWNERS 

### PR DESCRIPTION
**Pull request by rrsettgast**
This PR adds the CODEOWNERS file which will be used for enforcing reviews by specific individuals (code owners) on specific subdirectories. 

@castelletto1 @CusiniM @herve-gross @TotoGaz @paveltomin please look at the file and submit changes.

### CODE REVIEW

File: .github/CODEOWNERS
Comment (CusiniM) [inline .github/CODEOWNERS]: maybe this is not necessary since the `LvArray` PR will be the one actually containing the changes?

File: .github/CODEOWNERS
Comment (CusiniM) [inline .github/CODEOWNERS]:
```suggestion
/src/coreComponents/physicsSolvers/fluidFlow          @CusiniM @paveltomin 
```

File: .github/CODEOWNERS
Comment (CusiniM) [inline .github/CODEOWNERS]:
```suggestion
/src/coreComponents/constitutive                      @rrsettgast @CusiniM @dkachuma 
```
?

File: .github/CODEOWNERS
Comment (CusiniM) [inline .github/CODEOWNERS]:
```suggestion
/src/coreComponents/linearAlgebra                     @rrsettgast @castelletto1 @victorapm 
```

File: .github/CODEOWNERS
Comment (CusiniM) [inline .github/CODEOWNERS]: @paveltomin @castelletto1 ?

**Review by CusiniM — APPROVED** [review]
I am just suggesting a few additions. I wonder if we should always have at least 2 ppl so that we are never blocked by someone's absence (e.g., someone goes on vacation for 2 weeks).

File: .github/CODEOWNERS
Comment (rrsettgast) [reply .github/CODEOWNERS]: yeah...but sometimes people end up accidentally changing the hash

File: .github/CODEOWNERS
Comment (rrsettgast) [reply .github/CODEOWNERS]: just add it

File: .github/CODEOWNERS
Comment (rrsettgast) [inline .github/CODEOWNERS]:
```suggestion
/src/coreComponents/LvArray                           @rrsettgast @wrtobin @corbett5 @CusiniM 
```

File: .github/CODEOWNERS
Comment (rrsettgast) [inline .github/CODEOWNERS]:
```suggestion
/src/coreComponents/events                            @rrsettgast @cssherman 
```

File: .github/CODEOWNERS
Comment (CusiniM) [inline .github/CODEOWNERS]:
```suggestion
/src/coreComponents/physicsSolvers/multiphysics       @CusiniM @paveltomin @castelletto1 
```

File: .github/CODEOWNERS
Comment (paveltomin) [reply .github/CODEOWNERS]: > @paveltomin @castelletto1 ?

looks good
i would refine constitutive and multiphycs solvers a bit more if possible but that can be done later

File: .github/CODEOWNERS
Comment (MelReyCG) [inline .github/CODEOWNERS]: We discussed of who we would like to propose as potential code owners, and as the GEOS integration team at TE mission is to connect GEOS with user workflows, we would to propose @untereiner and @MelReyCG on these folders:
```
/src/coreComponents/codingUtilities                   @rrsettgast @untereiner
/src/coreComponents/common                            @rrsettgast @MelReyCG
/src/coreComponents/dataRepository                    @rrsettgast @wrtobin @MelReyCG
/src/coreComponents/fileIO                            @rrsettgast @wrtobin @MelReyCG @untereiner
/src/coreComponents/mesh                              @rrsettgast @wrtobin @untereiner
/src/coreComponents/schema                            @rrsettgast @untereiner
```

Comment (CusiniM) [reply .github/CODEOWNERS]: actually, can we exclude `/src/coreComponents/schema `? It's autogenerated and every time someone modifies even just a wrapper description it will be modified. So basically the codeowners of this folder will basically have to approve 95% of the PRs.

File: .github/CODEOWNERS
Comment (CusiniM) [inline .github/CODEOWNERS]:
```suggestion
/src/coreComponents/denseLinearAlgebra                @rrsettgast @CusiniM @castelletto1 
```

File: .github/CODEOWNERS
Comment (rrsettgast) [inline .github/CODEOWNERS]:
```suggestion
!/src/coreComponents/schema
```

File: .github/CODEOWNERS
Comment (MelReyCG) [reply .github/CODEOWNERS]: @CusiniM @rrsettgast 
And about `/src/coreComponents/dataRepository`? Is it possible to add me later on it (as I may contribute quite extensively on this folder in the future).

**Comment by paveltomin** [comment]
@rrsettgast is it possible to disable that for drafts?

**Comment by MelReyCG** [comment]
@paveltomin isn't it already "silenced" for draft? I think we only see the codeowners on drafts, but don't actually send notifications.

**Comment by paveltomin** [comment]
> @paveltomin isn't it already "silenced" for draft? I think we only see the codeowners on drafts, but don't actually send notifications.

I am not sure if it is silenced or not, sorry
I just saw people getting added as reviewers for drafts and got confused if that will trigger notifications for them

**Pull request merged**
